src/main/scala/nl.knaw.dans.easy.bagstore/server/StoresServletComponent.scala

```diff
@@ -147,21 +147,28 @@ trait StoresServletComponent extends DebugEnhancedLogging {
       trace(())
       basicAuth()
       debug("Authenticated")
-      val bagstore = params("bagstore")
+      val bagStore = params("bagstore")
       val uuidStr = params("uuid")
-      logger.info(s"Received PUT request for bagstore $bagstore and UUID $uuidStr")
-      bagStores.getBaseDirByShortname(bagstore)
+      val contentType = Option(request.getHeader("Content-Type"))
+      logger.info(s"Received PUT request for bagstore $bagStore and UUID $uuidStr")
+      bagStores.getBaseDirByShortname(bagStore)
         .map(base => {
           Try { UUID.fromString(uuidStr) }
             .recoverWith {
               case _: IllegalArgumentException => Failure(new IllegalArgumentException(s"invalid UUID string: $uuidStr"))
             }
+            .flatMap(uuid => //TODO make a method in
+              if (contentType.fold(true)(!_.equalsIgnoreCase("application/zip")))
```

Comment:
I'm not completely certain, but this if-else construction sounds a lot like:

```scala
contentType.withFilter(_.equalsIgnoreCase("application/zip"))
    .map(_ => uuid)
    .getOrElse(Failure(UnsupportedMediaTypeException(contentType.getOrElse("none"), "application/zip")))
```

Please verify with tests though 😉

Reply:
`.flatMap(uuid =>
              requestContentType.withFilter(_.equalsIgnoreCase("application/zip"))
                .map(_ => Success(uuid))
                .getOrElse(Failure(UnsupportedMediaTypeException(requestContentType.getOrElse("none"), "application/zip")))
            )`
This compiles and works

Reply:
The code looks much nicer now 👍

Reply:
extract method?